OptimoveSDK/optimove-sdk/src/main/java/com/optimove/android/optimobile/OptimobileHttpClient.java

```diff
@@ -0,0 +1,82 @@
+package com.optimove.android.optimobile;
+
+import android.os.Build;
+
+import java.io.IOException;
+import java.util.Collections;
+
+import okhttp3.Callback;
+import okhttp3.ConnectionSpec;
+import okhttp3.MediaType;
+import okhttp3.OkHttpClient;
+import okhttp3.Request;
+import okhttp3.RequestBody;
+import okhttp3.Response;
+
+class OptimobileHttpClient {
+    private final OkHttpClient okHttpClient;
+
+    OptimobileHttpClient() {
+        okHttpClient = this.buildOkHttpClient();
+    }
+
+    private OkHttpClient buildOkHttpClient() {
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.M) {
+            return new OkHttpClient();
+        }
+
+        //ciphers available on Android 4.4 have intersections with the approved ones in MODERN_TLS, but the intersections are on bad cipher list, so,
+        //perhaps not supported by CloudFlare. On older devices allow all ciphers
+        ConnectionSpec spec = new ConnectionSpec.Builder(ConnectionSpec.MODERN_TLS)
+                .allEnabledCipherSuites()
+                .build();
+
+        return new OkHttpClient.Builder()
+                .connectionSpecs(Collections.singletonList(spec))
+                .build();
+    }
+
+    Response postSync(String url, String data) throws IOException, Optimobile.PartialInitialisationException {
+        RequestBody body = RequestBody.create(data, MediaType.parse("application/json; charset=utf-8"));
+
+        Request.Builder builder = new Request.Builder().post(body);
+        Request request = this.buildRequest(builder, url);
+
+        return this.doSyncRequest(request);
+    }
+
+    Response getSync(String url) throws IOException, Optimobile.PartialInitialisationException {
+        Request.Builder builder = new Request.Builder().get();
+
+        Request request = this.buildRequest(builder, url);
+
+        return this.doSyncRequest(request);
+    }
+
+    void getAsync(String url, Callback callback) throws Optimobile.PartialInitialisationException {
+        Request.Builder builder = new Request.Builder().get();
+        Request request = this.buildRequest(builder, url);
+
+        this.doAsyncRequest(request, callback);
+    }
+
+    private Request buildRequest(Request.Builder builder, String url) throws Optimobile.PartialInitialisationException {
```

Comment:
Consider extracting the request builder and using it as a dependency.

Reply:
i can see
- benefits: can mock in tests, can do `doSyncRequest` instead of separate methods for post/get, more control of what gets into request
- drawbacks: couples consumers to okHttp (now they already handle Response, but even more), some repeating bits like building body would move to consumers, responsibility of building request is spread across 2 classes 

idk, to me this feels like we can refactor according to needs when we have more uses of it

Reply:
One day, this code will be covered by unit tests. There is a good rule in software development similar to the "boy scout rule" which says "Always leave the campground cleaner than you found it". You have already applied this rule by adding the new http_client, which has improved the codebase 🙂

Reply:
my rule is minimal effort rule 💀  Yes, one day we could add tests. Or 1 day world ends. And not clear what happens 1st :) If tests happen 1st we can same way as now pull builder up (if we end up needing to mock it at all, mb we assert requests sent to mocked client). 

Same way we can infinitely add good things which we may need in future. But 1) when we need them we have more info of what we need, some refactor would likely be necessary anyway 2) 80% of kumulos codebase is now removed -- any good code has lifespan

ofc debatable where the line goes, im not against improving the campground 🤐